File: mobile/app/src/main/java/com/example/signlink/data/models/kuis/Request.kt
package com.example.signlink.data.models.kuis

import com.google.gson.annotations.SerializedName

data class KuisReq(
    val name: String,
    @SerializedName("batas_waktu")
    val batasWaktu: Int,
    @SerializedName("soal_kuis")
    val soalKuis: List<SoalKuisReq>
)

data class SoalKuisReq(
    val soal: String,
    @SerializedName("jawaban_benar")
    val jawabanBenar: String,
    @SerializedName("opsi_kuis")
    val opsiKuis: List<OpsiKuisReq>
)

data class OpsiKuisReq(
    val label: String,
    val text: String
)
data class StatsKuisReq(
    @SerializedName("kuis_id")
    val latihanId: String,
    val score: Int
)

File: back-end/domains/kuis/request.go
package kuis

type KuisReq struct {
	Name       string        `json:"name" form:"name" binding:"required"`
	BatasWaktu int           `json:"batas_waktu" form:"batas_waktu" binding:"required"`
	SoalKuis   []SoalKuisReq `json:"soal_kuis" form:"soal_kuis" binding:"required"`
}

type SoalKuisReq struct {
	Soal         string        `json:"soal" form:"soal" binding:"required"`
	JawabanBenar string        `json:"jawaban_benar" form:"jawaban_benar" binding:"required"`
	OpsiKuis     []OpsiKuisReq `json:"opsi_kuis" form:"opsi_kuis" binding:"required"`
}

type OpsiKuisReq struct {
	Label string `json:"label" form:"label" binding:"required"`
	Text  string `json:"text" form:"text" binding:"required"`
}

type DeleteKuisReq struct {
	ID string `json:"id" binding:"required"`
}

type GetAllKuisReq struct {
	Page  int64
	Limit int64
}

type StatsKuisReq struct {
	KuisID string `json:"kuis_id" binding:"required"`
	Score  int    `json:"score" binding:"required"`
}
